homestar-runtime/src/settings.rs

```diff
@@ -133,3 +266,42 @@ impl Settings {
         s.try_deserialize()
     }
 }
+
+#[cfg(test)]
+mod test {
+    use crate::Settings;
+
+    #[test]
+    fn import_existing_key() {
+        let settings = Settings::build("fixtures/settings-import-ed25519.toml".into()).unwrap();
+
+        let msg = b"foo bar";
+        let signature = libp2p::identity::Keypair::ed25519_from_bytes([0; 32])
+            .unwrap()
+            .sign(msg)
+            .unwrap();
+        // round-about way of testing since there is no Eq derive for keypairs
+        assert!(settings
+            .node
+            .network
+            .keypair_config
+            .keypair()
+            .unwrap()
+            .public()
+            .verify(msg, &signature));
+    }
+
+    #[test]
+    fn import_secp256k1_key() {
+        let settings = Settings::build("fixtures/settings-import-secp256k1.toml".into()).unwrap();
+
+        settings.node.network.keypair_config.keypair().unwrap();
```

Comment:
minor: 

```suggestion
assert!(settings.node.network.keypair_config.keypair().is_ok())
```

Reply:
panic on unwrap arguably give better info on failure than an assert of `is_ok` here since the unwrap will print out the actual error and `is_ok` will just print the bool

Reply:
@mriise thinking about it, then `expect` is what we'd want (as it will also print the actual error, and the context for the error given).